oligo_designer_toolsuite/database/_oligo_database.py

```diff
@@ -721,58 +575,140 @@ def get_sequence_oligoid_mapping(
-        """Update attributes of oligonucleotides in the database with new values.
+        return attribute_value
 
-        Iterates through each oligonucleotide in the database and updates its attributes based on the provided new attribute values.
-        The update is done in place, modifying the existing attributes of the oligonucleotides.
+    ############################################
+    # Manipulation Functions
+    ############################################
 
-        Input format of new attribute:
-        new_attribute = {"region_x::1": {"attribute_y": 110}}
+    def update_oligo_attributes(self, new_oligo_attribute: dict) -> None:
 
-        :param new_oligo_attribute: A dictionary mapping oligonucleotide IDs to their new attributes.
-        :type new_oligo_attribute: dict
-        """
         for region_id, database_region in self.database.items():
             for oligo_id, oligo_attributes in database_region.items():
-                oligo_attributes.update(new_oligo_attribute[oligo_id])
+                if oligo_id in new_oligo_attribute:
+                    oligo_attributes.update(format_oligo_attributes(new_oligo_attribute[oligo_id]))
+
+    def filter_database_by_region(self, remove_region: bool, region_ids: Union[str, List[str]]) -> None:
+
+        # Check formatting
+        region_ids = check_if_list(region_ids)
+        if self.database:
+            for region_id in self.database.keys():
+                if remove_region and (region_id in region_ids):
+                    del self.database[region_id]
+                elif not remove_region and (region_id not in region_ids):
+                    del self.database[region_id]
+        else:
+            raise ValueError(
+                "Can not filter. Database is empty! Call the method load_database() or load_database_from_fasta() first."
+            )
+
+    def filter_database_by_oligo(self, remove_region: bool, oligo_ids: Union[str, List[str]]) -> None:
+
+        # Check formatting
+        oligo_ids = check_if_list(oligo_ids)
+        if self.database:
+            for region_id in self.database.keys():
+                oligo_ids_region = list(self.database[region_id].keys())
+                for oligo_id in oligo_ids_region:
+                    if remove_region and (oligo_id in oligo_ids):
+                        del self.database[region_id][oligo_id]
+                    elif not remove_region and (oligo_id not in oligo_ids):
+                        del self.database[region_id][oligo_id]
+        else:
+            raise ValueError(
+                "Can not filter. Database is empty! Call the method load_database() or load_database_from_fasta() first."
+            )
+
+    def filter_database_by_attribute_threshold(
+        self, attribute_name: str, attribute_thr: float, remove_if_smaller_threshold: bool
+    ) -> None:
+        oligos_to_delete = []
+        for region_id in self.database.keys():
```

Comment:
The code could be parallelized region-wise.

Reply:
True, we will address it in the oligo attribute issue